src/Npgsql/NpgsqlDataAdapter.cs

```diff
@@ -137,6 +140,74 @@ protected override void OnRowUpdating(RowUpdatingEventArgs value)
             get => (NpgsqlCommand?)base.InsertCommand;
             set => base.InsertCommand = value;
         }
+
+        internal Task<int> Fill(DataTable dataTable, bool async, CancellationToken cancellationToken = default)
+        {
+            if (async)
```

Comment:
nit: can use the conditional operator, and make this into an expression-bodied method.